### ISSUE TITLE
[Bug]: Dashboard empty/only displays background

### ISSUE THREAD

**Issue by jjourdin** [netdata/netdata#14129]
### Bug description

This is exactly the same issue as #10886

When using netdata with TLS, a connection to the dashboard only display the background image (no chart, no menu...).

Enabling compression as suggested in #10886 (by default in my configuration) does not help.

The following JS scripts appear to be truncated when viewing it from the web browser :
 - /static/js/2.92ca8446.chunk.js
 - /static/js/main.7d1bdca1.chunk.js

Everything else seems to be ok (for instance, /api/v1/charts displays a complete and valid JSON content)

### Expected behavior

Dashboard should display all exepected elements, menus, charts, etc

### Steps to reproduce

1. fresh-install netdata from source (https://github.com/netdata/netdata/releases/download/v1.37.1/netdata-latest.tar.gz)
2. Generate a x509 certificate and configure TLS 1.3 in netdata.conf to listen on TCP port 443
3. Restart the service
4. Connect to https://<node>
5. Check the output of the webconsole : "Loading failed for the <script> with source..."

### Installation method

from source

### System info

```shell
# uname -a
FreeBSD <hidden> 13.1-STABLE-HBSD FreeBSD 13.1-STABLE-HBSD #568  hardened/13-stable/master-n191566-a3afff79a90: Sat Nov 26 12:43:41 EST 2022     root@ci-01.md.hardenedbsd.org:/usr/obj/src/13-stable/amd64.amd64/sys/HARDENEDBSD amd64

# uname -K
1301509

/etc/os-release:NAME=FreeBSD
/etc/os-release:VERSION="13.1-STABLE-HBSD"
/etc/os-release:VERSION_ID="13.1"
```

### Netdata build info

```shell
Version: netdata v1.37.1
Configure options:  '--prefix=/opt/netdata/usr' '--sysconfdir=/opt/netdata/etc' '--localstatedir=/opt/netdata/var' '--libexecdir=/opt/netdata/usr/libexec' '--libdir=/opt/netdata/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--disable-dependency-tracking' '--with-bundled-protobuf' 'CFLAGS=-O2 -pipe' 'LDFLAGS='
Install type: custom
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES
    ACLK:                       YES
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         YES
Libraries:
    protobuf:                YES (bundled)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: NO
    CUPS:                    NO
    EBPF:                    NO
    IPMI:                    NO
    NFACCT:                  NO
    perf:                    NO
    slabinfo:                NO
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: NO
Debug/Developer Features:
    Trace Allocations:       NO
```

**Comment by Dim-P** [comment]
@jjourdin Are you getting the same error in `/var/log/netdata/error.log`: 

` 2022-12-07 12:57:23: netdata ERROR : WEB_SERVER[static2] : SSL_write() returned -1 bytes, SSL error 3 (
errno 35, Resource temporarily unavailable)
`

@underhood @MrZammler Do you guys know if this is related to https://github.com/netdata/netdata/pull/14026 ?

**Comment by MrZammler** [comment]
@Dim-P it could...

@jjourdin or @Dim-P can you try with 1.36.1 and check if you get the same behaviour?

@jjourdin Is the request being made from a remote machine? I.e. the browser is not on the same host as netdata, right?

**Comment by Habetdin** [comment]
I have something similar with my setup of 2 netdata instances.

Both of them use Ubuntu packages, running netdata version 1.37.1.

Problem: Parent Netdata dashboard [can't show most of the charts](https://imgur.com/a/yBMPUOr), while the Netdata Cloud dashboard can show them all.

The parent node has lots of `SSL error 1`:
```
2022-12-21 00:37:35: netdata INFO  : WEB_SERVER[static6] : clients wants to STREAM metrics.
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35426] : thread created with task id 460604
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35426] : set name of thread 460604 to STREAM_RECEIVER
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35426] : STREAM CHILD_NODE_HOSTNAME [<CHILD_NODE_IP>]:35426: receive thread created (task id 460604)
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35426] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35426]: initializing communication...
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35426] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35426]: established link with negotiated capabilities: VCAPS HLABELS CLAIM CLABELS COMPRESSION FUNCTIONS REPLICATION BINARY 
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35426] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35426]: receiving metrics...
2022-12-21 00:37:35: netdata ERROR : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35426] : SSL_read() returned 0 bytes, SSL error 1
2022-12-21 00:37:35: netdata ERROR : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35426] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35426]: disconnected (completed 8 updates).
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35426] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35426]: receive thread ended (task id 460604)
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35426] : thread with task id 460604 finished
2022-12-21 00:37:35: netdata INFO  : WEB_SERVER[static6] : clients wants to STREAM metrics.
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35440] : thread created with task id 460605
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35440] : set name of thread 460605 to STREAM_RECEIVER
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35440] : STREAM CHILD_NODE_HOSTNAME [<CHILD_NODE_IP>]:35440: receive thread created (task id 460605)
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35440] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35440]: initializing communication...
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35440] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35440]: established link with negotiated capabilities: VCAPS HLABELS CLAIM CLABELS COMPRESSION FUNCTIONS REPLICATION BINARY 
2022-12-21 00:37:35: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35440] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35440]: receiving metrics...
2022-12-21 00:37:36: netdata ERROR : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35440] : SSL_read() returned 0 bytes, SSL error 1
2022-12-21 00:37:36: netdata ERROR : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35440] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35440]: disconnected (completed 41 updates).
2022-12-21 00:37:36: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35440] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35440]: receive thread ended (task id 460605)
2022-12-21 00:37:36: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35440] : thread with task id 460605 finished
2022-12-21 00:37:36: netdata INFO  : WEB_SERVER[static3] : clients wants to STREAM metrics.
2022-12-21 00:37:36: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35446] : thread created with task id 460606
2022-12-21 00:37:36: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35446] : set name of thread 460606 to STREAM_RECEIVER
2022-12-21 00:37:36: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35446] : STREAM CHILD_NODE_HOSTNAME [<CHILD_NODE_IP>]:35446: receive thread created (task id 460606)
2022-12-21 00:37:36: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35446] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35446]: initializing communication...
2022-12-21 00:37:36: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35446] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35446]: established link with negotiated capabilities: VCAPS HLABELS CLAIM CLABELS COMPRESSION FUNCTIONS REPLICATION BINARY 
2022-12-21 00:37:36: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35446] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35446]: receiving metrics...
2022-12-21 00:37:37: netdata ERROR : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35446] : SSL_read() returned 0 bytes, SSL error 1
2022-12-21 00:37:37: netdata ERROR : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35446] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35446]: disconnected (completed 32 updates).
2022-12-21 00:37:37: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35446] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35446]: receive thread ended (task id 460606)
2022-12-21 00:37:37: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35446] : thread with task id 460606 finished
2022-12-21 00:37:37: netdata INFO  : WEB_SERVER[static4] : clients wants to STREAM metrics.
2022-12-21 00:37:37: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35454] : thread created with task id 460608
2022-12-21 00:37:37: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35454] : set name of thread 460608 to STREAM_RECEIVER
2022-12-21 00:37:37: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35454] : STREAM CHILD_NODE_HOSTNAME [<CHILD_NODE_IP>]:35454: receive thread created (task id 460608)
2022-12-21 00:37:37: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35454] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35454]: initializing communication...
2022-12-21 00:37:37: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35454] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35454]: established link with negotiated capabilities: VCAPS HLABELS CLAIM CLABELS COMPRESSION FUNCTIONS REPLICATION BINARY 
2022-12-21 00:37:37: netdata INFO  : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35454] : STREAM CHILD_NODE_HOSTNAME [receive from [<CHILD_NODE_IP>]:35454]: receiving metrics...
2022-12-21 00:37:37: netdata ERROR : STREAM_RECEIVER[CHILD_NODE_HOSTNAME,[<CHILD_NODE_IP>]:35454] : SSL_read() returned 0 bytes, SSL error 1
2022-12-21 00:37:37: netdata LOG FLOOD PROTECTION too many logs (201 logs in 11 seconds, threshold is set to 200 logs in 1200 seconds). Preventing more logs from process 'netdata' for 1189 seconds.
```

And the child node has lots of `SSL error 2`:
```
2022-12-21 00:39:11: netdata INFO  : PLUGINSD[nfacct] : STREAM CHILD_NODE_HOSTNAME [send]: sending metrics to parent...
2022-12-21 00:39:12: netdata ERROR : STREAM_SENDER[CHILD_NODE_HOSTNAME] : SSL_read() returned -1 bytes, SSL error 2 (errno 11, Resource temporarily unavailable)
2022-12-21 00:39:12: netdata ERROR : STREAM_SENDER[CHILD_NODE_HOSTNAME] : Clearing stream_collected_metrics flag in charts of host CHILD_NODE_HOSTNAME
2022-12-21 00:39:12: netdata ERROR : STREAM_SENDER[CHILD_NODE_HOSTNAME] : Clearing stream_collected_metrics flag in charts of host CHILD_NODE_HOSTNAME
2022-12-21 00:39:12: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME: attempting to connect to '<PARENT_NODE_IP>:20000' (default port: 20000)...
2022-12-21 00:39:12: netdata ERROR : PLUGIN[idlejitter] : STREAM CHILD_NODE_HOSTNAME [send]: not ready - collected metrics are not sent to parent.
2022-12-21 00:39:12: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: initializing communication...
2022-12-21 00:39:12: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: waiting response from remote netdata...
2022-12-21 00:39:12: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: established link with negotiated capabilities: VCAPS HLABELS CLAIM CLABELS COMPRESSION FUNCTIONS REPLICATION BINARY 
2022-12-21 00:39:12: netdata ERROR : STREAM_SENDER[CHILD_NODE_HOSTNAME] : Clearing stream_collected_metrics flag in charts of host CHILD_NODE_HOSTNAME
2022-12-21 00:39:12: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: enabling metrics streaming...
2022-12-21 00:39:12: netdata INFO  : PLUGINSD[nfacct] : STREAM CHILD_NODE_HOSTNAME [send]: sending metrics to parent...
2022-12-21 00:39:13: netdata ERROR : STREAM_SENDER[CHILD_NODE_HOSTNAME] : SSL_read() returned -1 bytes, SSL error 2 (errno 11, Resource temporarily unavailable)
2022-12-21 00:39:13: netdata ERROR : STREAM_SENDER[CHILD_NODE_HOSTNAME] : Clearing stream_collected_metrics flag in charts of host CHILD_NODE_HOSTNAME
2022-12-21 00:39:13: netdata ERROR : STREAM_SENDER[CHILD_NODE_HOSTNAME] : Clearing stream_collected_metrics flag in charts of host CHILD_NODE_HOSTNAME
2022-12-21 00:39:13: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME: attempting to connect to '<PARENT_NODE_IP>:20000' (default port: 20000)...
2022-12-21 00:39:13: netdata ERROR : PLUGIN[idlejitter] : STREAM CHILD_NODE_HOSTNAME [send]: not ready - collected metrics are not sent to parent.
2022-12-21 00:39:13: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: initializing communication...
2022-12-21 00:39:13: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: waiting response from remote netdata...
2022-12-21 00:39:13: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: established link with negotiated capabilities: VCAPS HLABELS CLAIM CLABELS COMPRESSION FUNCTIONS REPLICATION BINARY 
2022-12-21 00:39:13: netdata ERROR : STREAM_SENDER[CHILD_NODE_HOSTNAME] : Clearing stream_collected_metrics flag in charts of host CHILD_NODE_HOSTNAME
2022-12-21 00:39:13: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: enabling metrics streaming...
2022-12-21 00:39:13: netdata INFO  : PLUGINSD[nfacct] : STREAM CHILD_NODE_HOSTNAME [send]: sending metrics to parent...
2022-12-21 00:39:14: netdata LOG FLOOD PROTECTION too many logs (201 logs in 10 seconds, threshold is set to 200 logs in 1200 seconds). Preventing more logs from process 'netdata' for 1190 seconds.
```

Also, could be related to the problem:

Netdata started raising this alarm since the version 1.37 (I've never ever seen this alarm, at least since 2019):
```
🔴 PARENT_NODE is critical - 10min dbengine global flushing errors
netdata.dbengine_long_term_page_stats (dbengine)
10min dbengine global flushing errors
number of pages deleted due to failure to flush data to disk in the last 10 minutes. Metric data were lost to unblock data collection. To fix, reduce disk load or use faster disks.

✅ PARENT_NODE recovered - 10min dbengine global flushing errors
netdata.dbengine_long_term_page_stats (dbengine)
10min dbengine global flushing errors (was critical for 3 minutes and 30 seconds)
number of pages deleted due to failure to flush data to disk in the last 10 minutes. Metric data were lost to unblock data collection. To fix, reduce disk load or use faster disks.
```

**Comment by CreeperFace00** [comment]
I have also been dealing with similar issues for some time now, however at this point I am unsure whether it is related to SSL or not.
![image](https://user-images.githubusercontent.com/16279973/209695272-aa565d83-2a43-463c-8558-90e486a78fe8.png)
Looking at the dev tools it seems that it is returning 404 "No metrics where matched to query." when trying to get the data to fill the charts, this presumably being why the charts are blank.


Some of my charts also appear like this:
![image](https://user-images.githubusercontent.com/16279973/209698315-60a7161c-3671-4b58-b935-833aab3bf492.png)
Devtools show that it isn't even attempting to fetch metrics on this one and ones like it.

**Comment by Dim-P** [comment]
@jjourdin , @CreeperFace00 , @Habetdin  Could you please check if your issues have been fixed with the latest nightly release, since @MrZammler has merged a fix for this (https://github.com/netdata/netdata/pull/14120) ?

**Comment by Habetdin** [comment]
@Dim-P 

I've installed netdata v1.37.0-148-nightly (over 1.37.1) on both hosts, now I'm getting this in the child node log:

```
2023-01-17 16:52:39: netdata INFO  : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node: connecting to 'parent.netdata.node:20000' (default port: 20000)...
2023-01-17 16:52:39: netdata ERROR : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node [send to parent.netdata.node:20000]: remote node response is not understood, is it Netdata? - will retry in 60 secs, at 2023-01-17 16:53:39
2023-01-17 16:53:39: netdata INFO  : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node: connecting to 'parent.netdata.node:20000' (default port: 20000)...
2023-01-17 16:53:39: netdata ERROR : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node [send to parent.netdata.node:20000]: remote node response is not understood, is it Netdata? - will retry in 60 secs, at 2023-01-17 16:54:39
2023-01-17 16:53:43: netdata INFO  : ACLK_Main : RRDCONTEXT: received checkpoint command for claim id 'c141a88a-01b9-xxxx-9762-5452003ec23b', node id 'f6fea062-2d5b-xxxx-8c28-51061d59566e', while node 'child.netdata.node' has an active context streaming.
2023-01-17 16:53:43: netdata ERROR : ACLK_Main : RRDCONTEXT: received version hash 13409193782 for host 'child.netdata.node', does not match our version hash 344465. Sending snapshot of all contexts.
2023-01-17 16:54:39: netdata INFO  : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node: connecting to 'parent.netdata.node:20000' (default port: 20000)...
2023-01-17 16:54:40: netdata ERROR : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node [send to parent.netdata.node:20000]: remote node response is not understood, is it Netdata? - will retry in 60 secs, at 2023-01-17 16:55:40
2023-01-17 16:55:40: netdata INFO  : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node: connecting to 'parent.netdata.node:20000' (default port: 20000)...
2023-01-17 16:55:40: netdata ERROR : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node [send to parent.netdata.node:20000]: remote node response is not understood, is it Netdata? - will retry in 60 secs, at 2023-01-17 16:56:40
```

Here's output of `openssl s_client -connect parent.netdata.node:20000` from child node, the error is similar to the one I've posted in [#14120](https://github.com/netdata/netdata/pull/14120#issuecomment-1360387318):

```
CONNECTED(00000003)
depth=2 C = US, O = Internet Security Research Group, CN = ISRG Root X1
verify return:1
depth=1 C = US, O = Let's Encrypt, CN = R3
verify return:1
depth=0 CN = parent.node
verify return:1
---
Certificate chain
 0 s:CN = parent.node
   i:C = US, O = Let's Encrypt, CN = R3
   a:PKEY: id-ecPublicKey, 384 (bit); sigalg: RSA-SHA256
   v:NotBefore: Dec 16 00:00:00 2022 GMT; NotAfter: Mar 16 00:00:00 2023 GMT
 1 s:C = US, O = Let's Encrypt, CN = R3
   i:C = US, O = Internet Security Research Group, CN = ISRG Root X1
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Sep  4 00:00:00 2020 GMT; NotAfter: Sep 15 16:00:00 2025 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
...
-----END CERTIFICATE-----
subject=CN = parent.hostname
issuer=C = US, O = Let's Encrypt, CN = R3
---
No client certificate CA names sent
Peer signing digest: SHA384
Peer signature type: ECDSA
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 2900 bytes and written 402 bytes
Verification: OK
---
New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384
Server public key is 384 bit
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_256_GCM_SHA384
    Session-ID: 5CE93CF7C5984C0244FCDCE21F951C188FDC0C22083C7AA7EC1B58A4FCDBCA8D
    Session-ID-ctx:
    Resumption PSK: 5BACED8DAE4C5027DE2EB22EE83B02E4CB01927DBB3509E5CA8DBA8DD6988C1740C18CFCA3A7E1A759A576F2B38BE78A
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 7200 (seconds)
    TLS session ticket:
    0000 - 6f c6 00 ef 38 59 9e f0-dd 15 17 00 5d ef 0b 04   o...8Y......]...
    0010 - 59 f7 49 00 b4 5d ae 8d-bb de 96 b9 85 97 3f 7a   Y.I..]........?z
    0020 - 8c 3a 9b 89 7f 6b 7c a3-08 06 6a 38 75 2a 12 25   .:...k|...j8u*.%
    0030 - 87 1f a3 ef 1e 78 02 db-0b 2c e9 4c 25 01 5c 7a   .....x...,.L%.\z
    0040 - e0 32 4c e1 dc 37 00 6a-f4 f6 46 69 e2 bc 72 38   .2L..7.j..Fi..r8
    0050 - 2a 49 7d aa 5a b8 b5 33-43 74 8a f8 e0 5f cd 69   *I}.Z..3Ct..._.i
    0060 - fa 6d 7a 32 47 91 73 0a-80 21 b3 f4 bb 19 e8 19   .mz2G.s..!......
    0070 - 98 05 ad 6d 66 32 53 6c-55 1a 06 51 2f d1 23 f8   ...mf2SlU..Q/.#.
    0080 - e0 f8 9a 43 62 f4 4f 37-52 5d df 32 09 44 0c ac   ...Cb.O7R].2.D..
    0090 - a1 94 58 72 3c c3 6d 62-9f 7f b0 ef fc 85 da 92   ..Xr<.mb........
    00a0 - 54 43 9f 29 ed ee 07 e1-be b3 40 70 b1 15 dc 81   TC.)......@p....
    00b0 - b0 fe aa 88 67 bf 91 81-61 66 90 09 3f ae 5d 10   ....g...af..?.].
    00c0 - 14 97 53 bd ce 34 cc 4e-5f 66 5f 1c c8 25 90 99   ..S..4.N_f_..%..

    Start Time: 1673964050
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_256_GCM_SHA384
    Session-ID: AE829E6BDF1C4FF9A58472AE15C0280688BC4F4457CFF6CE7AAC9C8520910EBB
    Session-ID-ctx:
    Resumption PSK: 505BBA45E54B2421234C3F8CF0A7D58BECDB0B28B7A5CB9432C80FA150F3AA7960B8FA26F64266C37B5579CE8F2D20E6
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 7200 (seconds)
    TLS session ticket:
    0000 - 6f c6 00 ef 38 59 9e f0-dd 15 17 00 5d ef 0b 04   o...8Y......]...
    0010 - 30 c7 87 eb 51 55 7e 73-94 5a e1 f3 13 a8 af 03   0...QU~s.Z......
    0020 - 89 f0 16 53 1e 90 c0 68-1a 68 ec 29 ac 70 b2 b7   ...S...h.h.).p..
    0030 - 58 5e c7 92 f3 d9 2d 87-47 b9 b5 18 41 bf 95 d5   X^....-.G...A...
    0040 - 4f 4d 2e b3 e1 e3 cb 94-73 9a 51 5e 4e 83 63 72   OM......s.Q^N.cr
    0050 - 4a 35 1f 4c 48 00 c3 8f-3f c0 27 8d dd 92 ec c9   J5.LH...?.'.....
    0060 - 1d 78 7e 82 65 e3 ae ad-f3 ff 1a 2d 4f b9 30 ad   .x~.e......-O.0.
    0070 - 25 84 17 d5 6a c4 df a0-c4 4b f2 e5 62 1f eb 2c   %...j....K..b..,
    0080 - fb 81 cf 99 93 35 91 58-0f 43 5f c7 a9 9f 44 21   .....5.X.C_...D!
    0090 - ca 0d 21 93 c9 04 7e 9f-c7 20 50 9c 00 41 6d e8   ..!...~.. P..Am.
    00a0 - 2f 65 e3 c9 34 7a f6 2c-87 d2 10 e4 0a b5 8d 02   /e..4z.,........
    00b0 - ba ff d5 77 f7 32 95 45-eb f4 7c 06 a9 c5 8c 66   ...w.2.E..|....f
    00c0 - 6f dd c4 1e f9 57 cd 15-a9 fd 41 6f 95 bf 08 9d   o....W....Ao....

    Start Time: 1673964050
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
40B71C4CAD7F0000:error:0A000126:SSL routines:ssl3_read_n:unexpected eof while reading:../ssl/record/rec_layer_s3.c:308:
```

**Comment by Habetdin** [comment]
Hello, @MrZammler

No proxies in between, and port is 20000 because of the settings:
```
[web]
	bind to = unix:/run/netdata/netdata.sock *:20000=streaming^SSL=force
```

**Comment by MrZammler** [comment]
Hi @Habetdin can you try please with today's nightly, i.e. -158 ?

Thank you.

**Comment by Habetdin** [comment]
Hello again, @MrZammler

On netdata v1.37.0-158-nightly, I'm still getting streaming connection error on the child node:

```
2023-01-18 12:42:46: netdata INFO  : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node: connecting to 'parent.netdata.node:20000' (default port: 20000)...
2023-01-18 12:42:46: netdata ERROR : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node [send to parent.netdata.node:20000]: remote node response is not understood, is it Netdata? - will retry in 60 secs, at 2023-01-18 12:43:46
2023-01-18 12:43:46: netdata INFO  : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node: connecting to 'parent.netdata.node:20000' (default port: 20000)...
2023-01-18 12:43:46: netdata ERROR : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node [send to parent.netdata.node:20000]: remote node response is not understood, is it Netdata? - will retry in 60 secs, at 2023-01-18 12:44:46
```

I've also tried making SSL optional:

```
# on parent:
[web]
	bind to = unix:/run/netdata/netdata.sock *:20000=streaming^SSL=optional

# on child: removed the ":SSL" suffix from the stream `destination`
```

...and in this unencrypted state there's slightly different error:
```
2023-01-18 12:49:13: netdata INFO  : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node: connecting to 'parent.netdata.node:20000' (default port: 20000)...
2023-01-18 12:49:13: netdata ERROR : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node [send to parent.netdata.node:20000]: remote netdata does not respond.
2023-01-18 12:49:43: netdata INFO  : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node: connecting to 'parent.netdata.node:20000' (default port: 20000)...
2023-01-18 12:49:43: netdata ERROR : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node [send to parent.netdata.node:20000]: remote netdata does not respond.
```

May be there's some another issue that is not SSL-related?

**Comment by MrZammler** [comment]
Hmm, yes, let's try to remove SSL from the equation. Can you try to remove (rename for a bit) the ssl.key and ssl.perm you have? Just to make sure that the parent does not load them and does not support SSL at all. And then yes, let's try an unecrypted stream configuration to see.

Thanks!

**Comment by Habetdin** [comment]
@MrZammler I've commented out `ssl certificate` and `ssl key` options to cut the SSL off and after that child node successfully connected to the parent:

```
2023-01-18 20:40:43: netdata INFO  : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node: connecting to 'parent.netdata.node:20000' (default port: 20000)...
2023-01-18 20:40:43: netdata INFO  : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node [send to parent.netdata.node:20000]: established link with negotiated capabilities: VCAPS HLABELS CLAIM CLABELS COMPRESSION FUNCTIONS REPLICATION BINARY 
2023-01-18 20:40:43: netdata INFO  : STREAM_SENDER[child.netdata.node] : STREAM child.netdata.node [send to parent.netdata.node:20000]: enabling metrics streaming...
2023-01-18 20:40:43: netdata INFO  : PLUGIN[idlejitter] : STREAM child.netdata.node [send]: sending metrics to parent...
```

**Comment by MrZammler** [comment]
Hi @Habetdin 

Can you please send the `stream.conf`, `netdata.conf` from both parent and child to manolis@netdata.cloud? So I can try to replicate this? Thank you.

**Comment by Habetdin** [comment]
@MrZammler sure, just sent you the requested files.

**Comment by CreeperFace00** [comment]
It appears I am able to get streaming working again on troublesome nodes by doing the following:

1. Reset the netdata.public.unique.id by running ``sudo rm /var/lib/netdata/registry/netdata.public.unique.id``
2. Change the ``hostname =`` in ``netdata.conf`` to a new unique name
3. Restart the agent with ``sudo systemctl restart netdata``

This solved all of the streaming issues I was still encountering involving blank pages or empty charts.

**Comment by cakrit** [comment]
This solution of giving a child a new identity essentially sounds like a corrupted DB on the parent for that child, or something similar. @MrZammler perhaps we need to get the entire DB of the parent to figure it out?

**Comment by MrZammler** [comment]
Yes, will check with @CreeperFace00 . It could be similar though to https://github.com/netdata/netdata/issues/14293

**Comment by CreeperFace00** [comment]
I uploaded a copy of my database, it's pretty large at 27GB uncompressed. I compressed it down to 13GB
https://drive.google.com/file/d/1uPyyoHHO4aHmkR80UU6zdlz3v_0Wx6ub/view?usp=sharing

**Comment by MrZammler** [comment]
Hi! There have been a few changes that should address some of the issues here, including the SSL connectivity/streaming issues, as well as the case of having different guids from the same hostnames streaming.

We'll leave this open for a while if there is any feedback. Thanks!

**Comment by MrZammler** [comment]
Closing this.
